### ISSUE TITLE
[BUG] reconstruct from index number instead of  quant

### ISSUE THREAD

**Issue by eisneim** [TencentARC/Open-MAGVIT2#10]
for AR transformer we need the quantized index number to do generation task, so being able to reconstruct from index number is essential. 

I found a bug in the process of:  index -> quant -> decoder -> image pixel

 the LFQ.mask is initialized with reversed order: `[131072,  65536,  32768, ... 2, 1]`  but the code for  index -> quant is in normal order
https://github.com/TencentARC/Open-MAGVIT2/blob/3eaaa45d86976d27d57dbcf33465c137308ef74c/taming/modules/vqvae/lookup_free_quantize.py#L170

here is what i did, (BTW, this pretrained weights is only trained with 5 epoch on imagenet with higher compression: `256x256 -> 8x8` so the quality is really bad)

![reconstrct_openMagvit2_idx](https://github.com/TencentARC/Open-MAGVIT2/assets/5195894/3a9b733b-0b4c-40fc-8de4-5abb62f82b4c)

**Comment by RobertLuo1** [comment]
Thanks for your pointing out. Here for the first stage of tokenizer training, indices are not utilized. Therefore, it won't have any impact on the training. For the second stage of AR training, here indeed will have some influence. However, since it only defines a rule to map the bits to indices and [Open-MAGVIT2/taming/modules/vqvae/lookup_free_quantize.py](https://github.com/TencentARC/Open-MAGVIT2/blob/3eaaa45d86976d27d57dbcf33465c137308ef74c/taming/modules/vqvae/lookup_free_quantize.py#L170) here defines the whole codebook and use it for Entropy Loss calculation, it won't cause too much impact. Note that it is not our latest version of AR training code. In our latest version, we have already fixed it and designed something new. We will update the new code in the near future.
